### ISSUE TITLE
Optimize application reads with user-scoped caching

### ISSUE THREAD

**Issue by lurzar** [lurzar/expenses-app#35]
## Summary

Reduce repeated Planning queries by caching the authenticated user's planning collection without weakening ownership boundaries or making Redis a release requirement.

## Current context

- `PlanningService::getAllPlannings()` runs the same user-scoped query for Dashboard, Planning, and Expenses index pages.
- Planning create and delete are the only direct Planning mutations. Account delete and restore cascade to Planning records through `UserObserver`.
- The app currently defines file, array, Redis, and other Laravel cache stores. `.env.example` keeps `CACHE_DRIVER=file`, while Sail already provides Redis as an optional shared store.
- No application read currently uses `Cache`, `cache()`, cache tags, or a Planning cache abstraction.
- #61 still tracks single-record Planning ownership. This issue must not cache or expand those routes.

## Target outcome

Repeated list reads for one authenticated user reuse a short-lived cache entry. Successful mutations invalidate that user's entry after commit, and no user can receive another user's Planning data.

## Approved cache policy

- [x] Use Laravel's configured default cache store; keep `file` as the repository default and document Redis as an optional deployment choice.
- [x] Cache only `getAllPlannings()` in v2.0.9. Dashboard, Planning, and Expenses then share the same read-through entry.
- [x] Use a versioned, per-user key such as `planning:index:v1:user:{internal-user-id}`.
- [x] Use a five-minute TTL as a recovery boundary, not as the primary consistency mechanism.
- [x] Invalidate after successful Planning create/delete and after account delete/restore cascades.
- [x] Keep single-record reads, route-model binding, authentication, profiles, activity logs, Telescope data, validation failures, and error responses outside the cache.

## Implementation plan

- [ ] Add one small Planning cache boundary that owns key construction, TTL, read-through behavior, and invalidation.
- [ ] Keep authorization in the database query and authenticated route; the cache must never decide access.
- [ ] Invalidate only the affected user's key and preserve transaction semantics.
- [ ] Cache an empty collection so users without Planning records do not repeat the query.
- [ ] Keep the implementation compatible with file, array, and Redis stores without cache tags or store-specific APIs.
- [ ] Add focused tests before implementation, then refactor after they pass.

## Acceptance criteria

- [ ] The first index read queries the authenticated user's Planning records; a repeated read uses the cached value.
- [ ] User A and User B always use distinct keys and cannot receive each other's records.
- [ ] Empty collections are cached.
- [ ] Planning create and delete invalidate the affected user's collection only after a successful write.
- [ ] Account delete and restore invalidate the affected user's collection.
- [ ] A rolled-back mutation does not publish a new cache state.
- [ ] TTL expiry causes the next read to query the database and refresh the entry.
- [ ] The implementation does not require Redis, cache tags, a new UI/API, or changes to #61's ownership scope.
- [ ] Existing Planning, Dashboard, Expenses, profile deletion, and activity-log behavior remains intact.

## Validation plan

- Focused Pest tests cover cache miss/hit, empty results, user isolation, TTL expiry, mutation invalidation, and rollback behavior with the array store.
- Existing focused Planning, Expenses, Dashboard, Profile, and activity-log tests run against the release branch.
- Full Laravel and frontend validation follows the repository's known #65 baseline; unrelated inherited failures remain documented rather than hidden.
- A Redis smoke check may validate compatibility, but Redis is not required for the focused test suite.

## Risks and rollback

- Risk: a global or malformed key can expose financial Planning data across users.
- Risk: invalidating before commit or missing a cascade can serve stale records.
- Risk: store-specific behavior can make local and deployed environments diverge.
- Rollback: revert the cache layer and configuration. No schema or stored financial data migration is required.

## Dependencies and sequencing

- Release target: v2.0.9, aggregated into `v2.1`.
- Related security boundary: #61 remains independent and out of scope.
- Related quality baseline: #65 defines inherited full-suite and formatting work; it does not block focused cache tests.
- Documentation companion: #87 stacks above this implementation.
- Release tracker: #88.

## Branching and PR

- Base: `v2.0.9`
- Branch: `feat/35-add-user-scoped-planning-cache`
- PR: `Closes #35`

**Comment by lurzar** [comment]
Owner approval recorded on 2026-08-15:

- keep Laravel's configured default cache store and the repository's `file` default; Redis remains optional;
- cache only the authenticated user's Planning collection used by Dashboard, Planning, and Expenses indexes;
- use a versioned per-user key with a five-minute TTL;
- invalidate after successful Planning create/delete and account delete/restore cascades;
- exclude single-record reads, route-model binding, auth/profile data, activity logs, Telescope, validation failures, and error responses; and
- keep #64 independent while #87 stacks above #35.

Implementation may proceed test-first on `feat/35-add-user-scoped-planning-cache` from the `v2.0.9` release branch.

**Comment by lurzar** [comment]
Implementation is review-ready in PR #89.

- Branch/base: `feat/35-add-user-scoped-planning-cache` -> `v2.0.9`
- Focused cache suite: 10 passed, 33 assertions
- Full isolated backend suite: 67 passed, with the same five #65 baseline failures
- Changed-file Pint, Composer validation/audit, TypeScript, Vite build, and npm audit passed
- Complete five-file security diff review found no reportable issues

The implementation adds no schema, dependency, environment variable, UI, API, cache-tag requirement, or change to #61's single-record ownership scope. #87 will document the exact merged cache behavior as the stack's top layer.

**Comment by lurzar** [comment]
Completed in PR #89 and released through PR #93.

The v2.0.9 aggregate passed the focused Planning cache suite on both the array test store and the default file store: 10 tests, 33 assertions. Per-user isolation, empty-result caching, five-minute expiry, mutation invalidation, account lifecycle invalidation, unaffected-user preservation, and rollback behavior are covered. The lightweight `v2.0.9-dev` tag points to release commit `dc2ac69`.
